### PR TITLE
Adds Art Collecting: Less Broken Git Edition

### PR DESCRIPTION
**Pull request by Ivniinvi**
## About The Pull Request

Messed up git on #6895, this is the same thing.

Adds art collecting. There is now a verb in the OOC tab that lets you browse the art gallery. You can then buy any art piece from its owner for a price of 500 beecoins + 500 for every time it's changed hands. 1/4 of the sale price goes to the original artist, 1/4 goes to the person you bought it from, and 1/2 goes to the void. Additionally, if art is bought during the pregame, it sends a message in OOC describing the purchase. Each painting's initial owner is the artist.

Purchasing art is strictly nonprofitable, you will never make positive beecoins off of purchasing art. At best, you will make back slightly more than a quarter of the purchase price.

## Why It's Good For The Game
Good for artists because good art will change hands a lot, and send beecoins in their direction as an incentive for making good art.
Good for the players with a lot of beecoins because they can flex and build up a collection.
Good for the players with not many beecoins because it has zero (0) effect on the actual game.
Good for the game because it provides a large beecoin sink.

## Testing Photographs and Procedure
image
(Text is incorrect, and has since been corrected)

## Changelog
:cl: Ivniinvi
add: Art collecting! Use the Browse Art Gallery verb in the OOC tab to purchase art. The original artist gets 1/4 of the purchase price, and the current owner gets 1/4.
/:cl:

### CODE REVIEW

**Comment by mystery3525** [comment]
Also, another consideration I realized is how are we just going to prevent (if we will) someone from just copying art 😳 NFT reference

**Comment by Ivniinvi** [comment]
> Would it be better to have a split of 25% to artist, 50% to original owner, and 25% to the void? Or maybe less to void? I feel like if we have a Return on Investment of 1/4th the purchase price, this will cause prices for art to skyrocket after it changes a few hands and even the most well-endowed players can't buy certain art. I do agree a decent chunk should go to the original artist.

I think it would be best to start with the rates in the PR and edit if adjustment feels warranted.


> Also, another consideration I realized is how are we just going to prevent (if we will) someone from just copying art 😳 NFT reference

Currently, the only way to make art is actually decently time consuming and completely un-automatable, so there should be limited copying.

**Comment by mystery3525** [comment]
>I think it would be best to start with the rates in the PR and edit if adjustment feels warranted.

Fair enough

Would it be possible to add a date of creation/date of upload for the art? I feel this may add value to older pieces of art and perhaps combat any future issue of copying.

**Review by itsmeow — APPROVED** [review]
Looks fine surprisingly. How was this not reviewed earlier

**Comment by itsmeow** [comment]
Merge master, change get_metabalance() since it doesn't exist anymore

File: code/controllers/subsystem/persistence.dm [hunk 1]
```diff
@@ -294,6 +295,42 @@ SUBSYSTEM_DEF(persistence)
 	for(var/obj/structure/sign/painting/P in painting_frames)
 		P.load_persistent()
 
+	json_file = file("data/pending_art_metacoin.json")
+	if(fexists(json_file))
+		pending_art_metacoin = json_decode(file2text(json_file))
```
Comment (itsmeow) [inline code/controllers/subsystem/persistence.dm:329]: use rustg_file_read

File: code/controllers/subsystem/persistence.dm [hunk 2]
```diff
@@ -294,6 +295,42 @@ SUBSYSTEM_DEF(persistence)
...
+	. = ..()
+
+/datum/controller/subsystem/persistence/proc/process_art_payouts()
+	for(var/client/C in GLOB.clients)
+		if(pending_art_metacoin[C.ckey])
+			C.inc_metabalance(pending_art_metacoin[C.ckey], TRUE, "Art sales")
+			pending_art_metacoin -= C.ckey
+	var/json_file = file("data/pending_art_metacoin.json")
+	fdel(json_file)
+	WRITE_FILE(json_file, json_encode(pending_art_metacoin))
```
Comment (itsmeow) [inline code/controllers/subsystem/persistence.dm]: rustg_file_write

**Comment by itsmeow** [comment]
Add prettier extension or use
`cd tgui`
`yarn prettierx --write packages`

**Comment by itsmeow** [comment]
also suggest doing `git rebase -i master` to drop the incorrect commits

**Pull request closed without merging**